### PR TITLE
udev: open with O_NOCTTY

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/udev/fido_id/fido_id.c [hunk 1]
```diff
@@ -67,7 +67,7 @@ static int run(int argc, char **argv) {
         if (!desc_path)
                 return log_oom();
 
-        fd = open(desc_path, O_RDONLY | O_NOFOLLOW | O_CLOEXEC);
+        fd = open(desc_path, O_RDONLY | O_NOFOLLOW | O_CLOEXEC | O_NONBLOCK | O_NOCTTY);
```
Comment (poettering) [inline src/udev/fido_id/fido_id.c]: hmm, O_NONBLOCK? the effect of that highly depends on backing inode. Sometimes it has an effect on open() behaviour and sometimes it does not. It always has an effect on read() and so on. 

On tty devices O_NONBLOCK disables waiting for carrier iirc, and on some block devices it disables waiting for medium insert iirc. Here we open a sysfs attribute, no idea what effect this has, if any.

Generally: I think it makes sense to open block devices with O_NONBLOCk and some other devices, but usually you want to turn it off right after, so it doesn't affect read(). 

Or in other words, I am not entirely sure it's really the right thing to do here.

(O_NOCTTY is always OK I figure, if we might end up opening the wrong inode at some place we might end up with a contorlling terminal, which we don't want. Unlike O_NOCTTY has very clear meaning across all inodes, so it should be riskless to specify)

File: src/udev/udev-builtin-net_id.c [hunk 2]
```diff
@@ -229,23 +229,29 @@ static int dev_pci_onboard(sd_device *dev, const LinkInfo *info, NetNames *names
...
+        size_t len;
+        int r;
 
-        if (sd_device_get_syspath(dev, &syspath) < 0)
-                return false;
+        r = sd_device_get_syspath(dev, &syspath);
+        if (r < 0)
+                return r;
 
         filename = strjoina(syspath, "/config");
```
Comment (poettering) [inline src/udev/udev-builtin-net_id.c:242]: time for a sd_device_get_sysattr_path() or so?

File: src/udev/udev-builtin-net_id.c [hunk 3]
```diff
@@ -229,23 +229,29 @@ static int dev_pci_onboard(sd_device *dev, const LinkInfo *info, NetNames *names
...
-        fd = open(filename, O_RDONLY | O_CLOEXEC);
-        if (fd < 0)
-                return false;
-        if (read(fd, &config, sizeof(config)) != sizeof(config))
-                return false;
+        r = read_virtual_file(filename, PCI_HEADER_TYPE + 1, (char **) &config, &len);
```
Comment (poettering) [inline src/udev/udev-builtin-net_id.c:243]: or maybe instead an `sd_device_get_sysattr_value_binary()` or so

File: src/udev/fido_id/fido_id.c (hunk 1 above)
Comment (yuwata) [reply src/udev/fido_id/fido_id.c]: Interesting. Thank you for the explanation.

File: src/udev/udev-builtin-net_id.c (hunk 3 above)
Comment (yuwata) [reply src/udev/udev-builtin-net_id.c:243]: That sounds nice. But I'd like to leave this to another PR.

**Pull request merged**